src/main/scala/com/leagueprojecto/api/services/riot/RiotService.scala

```diff
@@ -42,34 +52,23 @@ trait RiotService {
     }
   }
 
-  protected def endpoint(regionParam: String, serviceParam: String, queryParams: Map[String, String] = Map.empty): Uri = {
-    region = regionParam
-    service = serviceParam
-
+  protected def riotGetRequest(regionParam: String, serviceParam: String, queryParams: Map[String, String] = Map.empty): Future[HttpResponse] = {
     val queryString = (queryParams + ("api_key" -> api_key)).collect { case x => x._1 + "=" + x._2 }.mkString("&")
-    val URL = s"/api/lol/$region/$service?$queryString"
+    val URL = s"/api/lol/$regionParam/$serviceParam?$queryString"
     log.debug(s"endpoint: $URL")
-    URL
+    Source.single(RequestBuilding.Get(URL)).via(riotConnectionFlow(regionParam, serviceParam)).runWith(Sink.head)
   }
 
-  protected def riotRequest(httpRequest: HttpRequest): Future[HttpResponse] =
-    Source.single(httpRequest).via(riotConnectionFlow).runWith(Sink.head)
+  protected def mapRiotTo[R](response: ResponseEntity, responseClass: Class[R]): Future[R] = {
 
-  protected def defaultSuccessHandler(origSender: ActorRef): PartialFunction[HttpResponse, Unit] = {
-    case HttpResponse(TooManyRequests, _, _, _) =>
-      val message = "Too many requests"
-      log.warning(message)
-      origSender ! Failure(new TooManyRequests(message))
+    val mappedResult: Future[String] = for {
```

Comment:
This for yield structure is probably not needed anymore since we are only doing one thing.